util/caller/resolver.go

```diff
@@ -55,57 +69,39 @@ func NewCallResolver(offset int, numStrip int) *CallResolver {
 		panic("could not look up callsite of NewCallResolver")
 	}
 	fileParts := strings.Split(file, "/")
-	cr.stripPrefix = strings.Join(fileParts[:len(fileParts)-numStrip-1], "/") + "/"
+	if len(fileParts) > numStrip {
+		cr.stripPrefix = strings.Join(fileParts[:len(fileParts)-numStrip-1], "/") + "/"
```

Comment:
This needs to work with more than just code in the cockroach package; we map raft's logging onto our own with a stack offset so these currently look like `I0701 17:46:03.569951   16075 src/github.com/coreos/etcd/raft/raft.go:390] 1 became follower at term 5`. 

I do generally like to have paths relative to the repo root, though. I wonder if it would be better to remove the first N directories (or remove a leading regex that understands the patterns of github/bitbucket/golang.org urls) instead of keeping the last N.